packages/migration-service/src/gen-transactions.ts

```diff
@@ -736,43 +743,43 @@ async function run() {
     /// Iterate through accounts in order so we don't create 1mm promises.
     for (const [address, account] of chunk) {
       const solAddress: PublicKey | undefined = toSolana(address);
-      const isRouter = routers.has(address);
+      const routers = ownersToRouters[address];
 
-      if (isRouter) {
+      if (routers) {
         const dcBal = new BN(account.dc);
         routerBalances.hnt = routerBalances.hnt.add(new BN(account.hnt));
         routerBalances.mobile = routerBalances.mobile.add(
           new BN(account.mobile)
         );
-        routerBalances.iot = routerBalances.iot.add(
-          new BN(account.iot)
-        );
+        routerBalances.iot = routerBalances.iot.add(new BN(account.iot));
         routerBalances.stakedHnt = routerBalances.stakedHnt.add(
           new BN(account.staked_hnt)
         );
-        const instruction = await dcProgram.methods
-          .genesisIssueDelegatedDataCreditsV0({
-            amount: dcBal,
-            routerKey: address,
-          })
-          .accounts({
-            dcMint: dc,
-            dataCredits,
-            lazySigner,
-            circuitBreaker: dcCircuitBreaker,
-            dao,
-            subDao,
-          })
-          .instruction();
+        for (const router of routers) {
+          const instruction = await dcProgram.methods
+            .genesisIssueDelegatedDataCreditsV0({
+              amount: dcBal.div(new BN(routers.length)),
```

Comment:
Is dividing evenly correct?

Reply:
Yeah, realistically this shouldn't even happen.